src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -310,8 +312,11 @@ export default function Swap({
   useEffect(() => {
     if (approvalState === ApprovalState.PENDING) {
       setApprovalSubmitted(true)
+    } else if (prevApprovalState === ApprovalState.PENDING && approvalState === ApprovalState.NOT_APPROVED) {
+      // user canceled the approval tx, reset the UI
+      setApprovalSubmitted(false)
```

Comment:
I guess you assume if it was pending, and is not anymore, then it was canceled. 

I wonder what would happen if you just, for example send an insufficient gasLimit. The tx is executed and reverted. Is it possible that this code have an issue with that case?